hack/envsubst.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/drone/envsubst"
```

Comment:
Why can not we just use shell `envsubst`?

Reply:
It's an option. We will need this in the builder image (which is debian based though), and we need a specific behaviour (= replace only the env vars which have an actual value set) which is a bit harder to guarantee using shell tools.
I think all in all this approach has enough merit to be proposed, but I'm not pushing for it so if we all agree to use the shell variant (provided the above requirements are met) we can go this way.